sentry_sdk/client.py

```diff
@@ -136,6 +147,50 @@ def __setstate__(self, state):
         self.options = state["options"]
         self._init_impl()
 
+    def _setup_instrumentation(self, functions_to_trace):
+        # type: (Sequence[Dict[str, str]]) -> None
+        """
+        Instruments the functions given in the list `functions_to_trace` with the `@sentry_sdk.tracing.trace` decorator.
+        """
+        for function in functions_to_trace:
+            class_name = None
+            function_qualname = function["qualified_name"]
+            module_name, function_name = function_qualname.rsplit(".", 1)
+
+            try:
+                # Try to import module and function
+                # ex: "mymodule.submodule.funcname"
+
+                module_obj = import_module(module_name)
+                function_obj = getattr(module_obj, function_name)
+                setattr(module_obj, function_name, trace(function_obj))
+
+            except module_not_found_error:
+                try:
+                    # Try to import a class
+                    # ex: "mymodule.submodule.MyClassName.member_function"
+
+                    module_name, class_name = module_name.rsplit(".", 1)
+                    module_obj = import_module(module_name)
```

Comment:
Yes, true. If there is code executed on module load it could be have unintentional behavior.

But this optional and is only active when `functions_to_trace` is set.

What if we point out this concerns in the documentation, so people know about them when activating this feature?

Reply:
I’m okay with importing from strings. But I can’t say about the feature as such.

Reply:
Are you saying you can not say anything about the feature, or are you saying that you are not okay with the feature @mitsuhiko ?

Reply:
(On another note, @mdtro (Matthew T) said in slack: A note in the documentation is sufficient, imo. The list should be static or loaded from a secure source (no user or untrusted input).)